analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -87,70 +86,55 @@ internal static class DiagnosticVerifier
             }
         }
 
-        public static IEnumerable<Diagnostic> GetDiagnostics(Compilation compilation,
-            DiagnosticAnalyzer diagnosticAnalyzer, CompilationErrorBehavior checkMode,
-            bool verifyNoExceptionIsThrown = true,
-            string sonarProjectConfigPath = null) =>
-            GetDiagnostics(compilation, new[] { diagnosticAnalyzer }, checkMode, verifyNoExceptionIsThrown, sonarProjectConfigPath);
-
-        public static void VerifyNoIssueReported(Compilation compilation,
-                                                 DiagnosticAnalyzer diagnosticAnalyzer,
-                                                 CompilationErrorBehavior checkMode = CompilationErrorBehavior.Default,
-                                                 string sonarProjectConfigPath = null) =>
-            GetDiagnostics(compilation, diagnosticAnalyzer, checkMode, sonarProjectConfigPath: sonarProjectConfigPath).Should().BeEmpty();
-
-        public static ImmutableArray<Diagnostic> GetAllDiagnostics(Compilation compilation,
-            IEnumerable<DiagnosticAnalyzer> diagnosticAnalyzers, CompilationErrorBehavior checkMode,
-            bool verifyNoException = true,
-            CancellationToken? cancellationToken = null,
+        public static void VerifyNoIssueReported(
+                Compilation compilation,
+                DiagnosticAnalyzer diagnosticAnalyzer,
+                CompilationErrorBehavior checkMode = CompilationErrorBehavior.Default,
+                string sonarProjectConfigPath = null) =>
+            GetDiagnosticsNoExceptions(compilation, diagnosticAnalyzer, checkMode, sonarProjectConfigPath: sonarProjectConfigPath).Should().BeEmpty();
+
+        public static IEnumerable<Diagnostic> GetDiagnosticsNoExceptions(
```

Comment:
I think it's nice that you split the old code into these 2 new methods. 

In my opinion, we should get the exception assertion outside of this class but as this code will be changed anyway it can also stay here for now. Up to you.